src/jdk.jlink/linux/classes/jdk/tools/jlink/internal/plugins/StripNativeDebugSymbolsPlugin.java

```diff
@@ -131,6 +136,33 @@ public ResourcePool transform(ResourcePool in, ResourcePoolBuilder out) {
         return out.build();
     }
 
+    /**
+     * Method to determine if a particular resource should be stripped.
+     *
+     * Particular paths are added here to handle libraries within the openjceplus module.
+     * The FIPS certified library located in the C/icc directory is sensitive to
+     * any modifications to the native library. Performing any modifications to the library
+     * in any way, causes the FIPS library to fail to load due to a self verification check made.
+     *
+     * @param resource the resource to examine for stripping eligibility.
+     * @return return true if stripping should be done on a particular resource, false otherwise.
+     */
+    private boolean needsStripTransformation(ResourcePoolEntry resource) {
+        if ((resource.type() == ResourcePoolEntry.Type.NATIVE_LIB &&
+            resource.path().endsWith(SHARED_LIBS_EXT)) ||
+            resource.type() == ResourcePoolEntry.Type.NATIVE_CMD) {
+            if (resource.moduleName().equals("openjceplus") &&
+                resource.path().endsWith(SHARED_LIBS_EXT) &&
+                resource.path().contains("C/icc")) {
+                return false;
+            } else {
+                return true;
+            }
+        } else {
+            return false;
+        }
```

Comment:
This method should be `static`. Suggest a `switch` statement makes this clearer and avoids redundant tests:
```java
        switch (resource.type()) {
        case NATIVE_CMD:
            return true;
        case NATIVE_LIB:
            String path = resource.path();
            if (path.endsWith(SHARED_LIBS_EXT)) {
                if (!(resource.moduleName().equals("openjceplus") && path.contains("/C/icc/"))) {
                    return true;
                }
            }
            break;
        default:
            break;
        }
        return false;
```
Note:
* this has exactly one call to `type()` and at most one call to `path()`
* the path "contains" test is more specific

Reply:
Updated / tested as suggested above.